content/docs/rfcs/51/README.md

```diff
@@ -173,45 +169,99 @@ This example node is part of shards `13`, `14`, and `45` in the Status main-net
-
-From an app point of view, a subscription to a content topic `waku2/xxx` using automatic sharding would look like:
-
-`subscribe("/waku2/xxx", auto=true)`
-
-The app is oblivious to the pubsub topic layer.
-(Future versions could deprecate the default pubsub topic and remove the necessity for `auto=true`.)
+Autosharding selects shards automatically and is the default behaviour for shard choice.
+The other choices being static and named sharding as seen in previous sections.
+Shards (pubsub topics) MUST be computed from content topics with the procedure below.
+
+## Rendezvous Hashing
+
+Also known as the Highest Random Weight (HRW) method, has many properties useful for shard selection.
+- Distributed agreement. Local only computation without communication.
+- Load balancing. Content topics are spread as equally as possible to shards.
+- Minimal disruption. Content topics switch shards infrequently.
+- Low overhead. Easy to compute.
+
+### Algorithm
+
+For each shard,
+hash using Sha2-256 the concatenation of
+the content topic `application` field (N UTF-8 bytes),
+`version` (N UTF-8 bytes),
+the `cluster` index (2 bytes) and
+the `shard` index (2 bytes)
+take the first 64 bits of the hash,
+divided by 2^64,
+compute the natural logarithm of this number then
+take the negative of the weight (default 1.0) divided by it.
+Finally, sort the shard value pairs.
+
+The shard with the highest value MUST be used.
+
+### Example
+| Field | Value | Hex |
+|---    |---    |---  |
+| `application`| "myapp"| 0x6d79617070|
+| `version`    | "1"    | 0x31
+| `cluster`    | 1      | 0x0001
+| `shard`      | 6      | 0x0006
+
+- SHA2-256 of `0x6d796170703100010006` is `0xfdac5fb315b791cca3ccde738ebb0b8d2742519fce1086f2a3d2409cb22a7dd2`
+- The first 64 bits `0xfdac5fb315b791cc` divided by `0xffffffffffffffff` is ~`0.99`
+- The natual logarithm of `0.99` is ~`-0.01`
+- `-1` divided by `-0.01` equals ~`99.5`
+- Shard 6 has the priority value 99.5
+
+## Content Topics Format for Autosharding
+Content topics MUST follow the format in [23/WAKU2-TOPICS](https://rfc.vac.dev/spec/23/#content-topic-format).
+In addition, 2 prefixes MAY be added to content topics.
+Generation & bias.
+When omitted default values are used.
+Generation default value is `0`.
+Bias default value is `unbiased`.
+
+### Example
+
+`/0/unbiased/myapp/1/mysub/cbor`
+
+### Generation
+The generation number monotonously increases and indirectly refers to the total number of shards of the Waku Network.
+
+The first generation (zero) MUST use 8 shards in total.
+The cluster index is 1 and the indices of each shards are numbered 0 to 7.
+
+This document will be updated for future generation.
+
+### Bias
+Bias is used to skew the priority of shards via weights.
+Other biases than `unbiased` are unspecified for now but may be used in the future.
+
+### Topic Design
+Content topics have 2 purposes filtering and routing.
+Filtering is done by changing the `subject` field.
```

Comment:
What `subject` field? :D https://rfc.vac.dev/spec/23/#content-topic-format refers to `{content-topic-name}`. We could consider changing to `{subject}`, but should be consistent. For now I think `{content-topic-name}` is OK.

Reply:
:thinking: where did I saw subject...